state/unit.go

```diff
@@ -130,7 +130,9 @@ func (u *Unit) ConfigSettings() (charm.Settings, error) {
 	}
 	result := chrm.Config().DefaultSettings()
 	for name, value := range settings.Map() {
-		result[name] = value
+		if value != nil {
```

Comment:
This should not happen but it's the correct way to treat nils - fall through default. The logic should be explicit.